request_test.go

```diff
@@ -306,3 +310,78 @@ func TestDetectEncoding(t *testing.T) {
 		assert.Equal(t, tt.want, got, "detectEncoding(%+v)", tt.opts)
 	}
 }
+
+func TestNewRequestWithMetadata(t *testing.T) {
+	req := &transport.Request{Method: "foo"}
+	topts := TransportOptions{ServiceName: "bar"}
+	req, err := prepareRequest(req, nil /* headers */, Options{TOpts: topts})
+	assert.NoError(t, err)
+	assert.Equal(t, "foo", req.Method)
+	assert.Equal(t, "bar", req.TargetService)
+}
+
+func TestNewRequestWithTransportMiddleware(t *testing.T) {
+	req := &transport.Request{Method: "foo"}
+	topts := TransportOptions{ServiceName: "bar"}
+	restore := transportmiddleware.Register(mockTransportMiddleware{})
+	defer restore()
+	req, err := prepareRequest(req, nil /* headers */, Options{TOpts: topts})
+	assert.NoError(t, err)
+	assert.Equal(t, "bar", req.Method)
+	assert.Equal(t, "bar", req.TargetService)
+}
+
+type mockTransportMiddleware struct{}
+
+func (tm mockTransportMiddleware) Apply(_ context.Context, req *transport.Request) (*transport.Request, error) {
```

Comment:
once we apply middleware after the request is created, verify the request properties here